### ISSUE TITLE
Incorrect Validation Set File and Potential Data Leakage in Provided Dataset Files

### ISSUE THREAD

**Issue by mailxxttmailxxtt-max** [arcaninez/ODM3D#4]
Dear authors,

Thank you for releasing this excellent work and open-sourcing the code—it has been very helpful!

While attempting to reproduce the reported results, I encountered some issues with the provided dataset files that appear to cause significant data leakage during evaluation.

### Problem 1: `kitti_infos_val.pkl` is identical to `kitti_infos_train_valid.pkl`
Both pickle files contain exactly the same 3,712 samples with identical image IDs, meaning the validation set is effectively the same as (part of) the training set. This results in 100% data leakage when evaluating on "val".

**Verification code** (Python):
```python
import pickle

with open('data/kitti/kitti_infos_train_valid.pkl', 'rb') as f:
    train_data = pickle.load(f)
with open('data/kitti/kitti_infos_val.pkl', 'rb') as f:
    val_data = pickle.load(f)

train_ids = set(info['image']['image_idx'] for info in train_data)
val_ids = set(info['image']['image_idx'] for info in val_data)

print(f"Train samples: {len(train_ids)}")
print(f"Val samples: {len(val_ids)}")
print(f"Identical sets: {train_ids == val_ids}")  # Output: True
Problem 2: Training set overlaps with the standard KITTI validation range
The provided kitti_infos_train_valid.pkl includes:

~1,863 samples from indices 000000–003711 (~50.2%)
~1,849 samples from indices 003712–007480 (~49.8%)

This means approximately 49.8% of the training data comes from the commonly used KITTI validation range (indices 003712–007480, ~3,769 samples). When evaluating on this standard val split, nearly half of the "test" samples have been seen during training, leading to severely inflated performance.
My Reproduced Results
When I evaluate the model on the standard KITTI validation set (indices 003712–007480, 3,769 samples):

Car 3D AP (Moderate): 41.30%
(compared to 23.84% reported in the paper Table 1)

The much higher score is consistent with the leakage described above.
Questions

Could you please provide the correct kitti_infos_val.pkl file corresponding to a proper validation split (e.g., the commonly used 3,769-sample val set from indices 003712–007480)?
Which validation set was used to obtain the results reported in the paper (Table 1: 35.09% Easy, 23.84% Moderate, etc.)?
Is it intentional for the training set (kitti_infos_train_valid.pkl) to include samples from the standard validation range (003712–007480)?

Thank you very much for your time and clarification—this will greatly help the community reproduce and build upon your strong results!
Best regards,
mailxxttmailxxtt@gmail.com

**Comment by arcanienz** [comment]
Hi, thanks for your interest in our work. Apologies for the incorrect data files. I'll look into my local disk to see if I can find the correct one. Meanwhile, you may also try producing your own data files as it is quite easy. When doing that, I followed https://github.com/Cc-Hy/CMKD/blob/main/docs/GETTING_STARTED.md, which may be helpful.

**Comment by mailxxttmailxxtt-max** [comment]
Hi @arcanienz,

Thank you for the prompt response and the suggestion.

I have actually tried generating the dataset myself following the CMKD
guide you mentioned, ensuring there is strictly no data leakage in the
validation set. However, I noticed that when training on this generated
"clean" dataset, the performance drops by about 2-3% compared to the
results reported in the paper.

Could you provide some insights on this? I am wondering if there are any
specific hyperparameters or configurations I should adjust when using the
standard generated data to match the paper's performance.

Thanks again for your help! Regards

**Comment by arcanienz** [comment]
Hi, sorry for the delay in response.

Unfortunately, as this project was completed long ago, I am unable recall some details or give you specific suggestions to examine the issue.

However, by using the clean dataset, you should be able to reproduce the paper results. This is because we had also reproduced CMKD on our dataset and ensured that our reproduced results are close to those in the CMKD paper. This means that the results in our paper are obtained on the correct clean dataset.

As a first step, I would suggest removing the CMAug first to see whether it matches the ablation results (w/o CMAug) reported in the paper.
